lxqtgridlayout.cpp

```diff
@@ -569,8 +571,9 @@ QSize GridLayout::sizeHint() const
     if (!d->mIsValid)
         const_cast<GridLayoutPrivate*>(d)->updateCache();
 
-    return {d->cols() * d->mCellSizeHint.width(),
-                 d->rows() * d->mCellSizeHint.height()};
+    const int sp = spacing();
+    return {qMax(0, d->cols() * (d->mCellSizeHint.width() + sp) - sp),
```

Comment:
How about this?
```diff
diff --git a/lxqtgridlayout.cpp b/lxqtgridlayout.cpp
index 417deb1..a16b48a 100644
--- a/lxqtgridlayout.cpp
+++ b/lxqtgridlayout.cpp
@@ -571,9 +571,11 @@ QSize GridLayout::sizeHint() const
     if (!d->mIsValid)
         const_cast<GridLayoutPrivate*>(d)->updateCache();
 
+    if (d->mVisibleCount == 0)
+      return {0, 0};
+
     const int sp = spacing();
-    return {qMax(0, d->cols() * (d->mCellSizeHint.width() + sp) - sp),
-            qMax(0, d->rows() * (d->mCellSizeHint.height() + sp) - sp)};
+    return {d->cols() * (d->mCellSizeHint.width() + sp) - sp, d->rows() * (d->mCellSizeHint.height() + sp) - sp};
 }
 
 
```

Reply:
That works too.

Reply:
> That works too.

:)... but what do you thing about performance of those two approaches?

Reply:
I doubt that there will be any significant difference in performance either way. We are probably looking at single digits of instruction counts.

Reply:
Please, use the suggested solution (`if (d->mVisible...` and to `qMax(...`) and squash the commits.